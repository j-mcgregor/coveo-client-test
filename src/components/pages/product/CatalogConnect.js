import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { getCatalogItems } from '../../../flux/actions/catalogActions';
import CatalogContainer from './CatalogContainer';

const mapStateToProps = state => ({
  products: state.catalog.products
});

const mapDispatchToProps = dispatch => ({
  getCatalogItems: query => dispatch(getCatalogItems({ query }))
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(CatalogContainer)
);
