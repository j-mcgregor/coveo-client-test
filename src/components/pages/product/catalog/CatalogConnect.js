/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable arrow-parens */
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { getCatalogItems } from '../../../../flux/actions/catalogActions';
import CatalogContainer from './CatalogContainer';

const mapStateToProps = state => ({
  products: state.catalog.products,
  loading: state.catalog.loading
});

const mapDispatchToProps = dispatch => ({
  getCatalogItems: (query, api, options = {}) =>
    dispatch(getCatalogItems({ query, api, options }))
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(CatalogContainer)
);
