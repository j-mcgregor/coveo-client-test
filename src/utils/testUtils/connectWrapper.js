import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

export default (initialState, component) => {
  const mockStore = configureStore([thunk]);
  const store = mockStore(initialState);
  const wrapper = mount(
    <Provider store={store}>
      <BrowserRouter>{component}</BrowserRouter>
    </Provider>
  );

  return wrapper;
};
