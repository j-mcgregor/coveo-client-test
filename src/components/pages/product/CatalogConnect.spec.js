import React from 'react';
import CatalogConnect from './CatalogConnect';
import connectWrapper from '../../../utils/testUtils/connectWrapper';

describe('CatalogConnect', () => {
  it('should render with the store', () => {
    const initialState = {
      catalog: {
        products: [],
        loading: false
      }
    };
    const wrapper = connectWrapper(initialState, <CatalogConnect />);

    expect(wrapper).toMatchSnapshot();
  });
});
