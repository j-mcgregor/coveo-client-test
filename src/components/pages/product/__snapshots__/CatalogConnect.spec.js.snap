// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogConnect should render with the store 1`] = `ReactWrapper {}`;
